lib/requester/requester.js

```diff
@@ -92,10 +92,24 @@ _.extend(Requester.prototype, /** @lends Requester.prototype */ {
      * @param {Function} callback
      */
     request: function (id, request, callback) {
-        var self = this,
-            cookieJar = self.options.cookieJar,
-            requestOptions = util.getRequestOptions(request, self.options),
-            startTime = Date.now();
+        var error,
+            self = this,
+            cookieJar,
+            requestOptions,
+            startTime;
+
+        // at this point the request could have come from collection, auth or sandbox
+        // we can't trust the integrity of this request
+        // bail out if request url is empty
+        if (!(request && request.url && request.url.toString())) {
```

Comment:
@kunagpal `toString` is on the `Object.prototype` which means it should be available on all types except `undefined` and `null` which `!request.url` handles. The only case where this might be otherwise is when someone does `delete url.toString` or `url.toString = null`. In that case, we may need to check `_.isFunction`.

Isn't that too much work on the critical path? I may be wrong as well.